### PR TITLE
Checkout application

### PR DESCRIPTION
**Pull request by dcDalin**
# What does this PR do?

Checkout Workflow
- Cart Summary page
- Checkout Credit Card Info page
- Order Confirmation page

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
-app.get('/assertions', (req, res) => {
-    res.sendFile(path.join(__dirname + '/src/assertions/index.html')); 
+app.get("/checkout", (req, res) => {
+	const { total } = req.query;
+	res.render("checkout", {
+		total
+	});
 });
 
-app.get('/diagnostic', (req, res) => {
```
Comment (ghost) [inline app.js:16]: I should have mentioned, I need this route for something else. Please do not delete it.

File: views/success.pug [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  title Confirmation
+nav.navbar.navbar-expand-lg.navbar-light.bg-light
+  .container
+    a.navbar-brand(href='/') Store
+
+section
+  .container
+    .row
+      .col-lg-4.col-md-6
+        h1 Thank you, we have received your order
```
Comment (ghost) [inline views/success.pug]: Would it be possible to get the order total and the shipping address on this page as well?

File: views/checkout.pug [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+              .col
+                input.form-control(type='text' placeholder='Last name')
+          .form-group
+            .row
+              .col
+                input.form-control(type='text' placeholder='Card Number')
+          .form-group
+            .row
+              .col
+                input.form-control(type='date' placeholder='Expiration Date')
```
Comment (ghost) [inline views/checkout.pug]: Credit card expiration date should be just month and year, no day

**Review by ghost — CHANGES_REQUESTED** [review]
Great first pass @dcDalin . A couple features that I'd still like to see on the checkout page:
1. Billing Address fields
2. Shipping address fields that disappear if you select "same as billing address"

File: app.js (hunk 1 above)
Comment (dcDalin) [reply app.js:16]: Okay, fixing it now.

File: views/success.pug (hunk 2 above)
Comment (dcDalin) [reply views/success.pug]: Yes it is.

**Review by ghost — APPROVED** [review]

**Pull request merged**